스키마(http/https)가 없는 링크가 생성되어 CI 빌드를 깨뜨림

## 버그 설명
jekyll로 빌드된 _site 폴더에서 linkpreview 플러그인이 적용된 포스트의 html파일에 스키마(http/https)가 없는 링크가 생성되어 Git Actions의 빌드를 깨드림

## 재현 경로
linkpreview 플러그인이 적용된 2026-01-21-spring-security-contrib.md가 Git Actions에서 빌드될 경우

## 현재 동작
build status Failure

빌드의 _site 테스트 단계에서 문제 발생 
```
Run bundle exec htmlproofer _site \
Running 3 checks (Images, Links, Scripts) in ["_site"] on *.html files ...


Checking 27 internal links
Checking internal link hashes in 1 file
Ran on 9 files!
For the Links check, the following failures were found:

* At _site/posts/spring-security-contrib/index.html:13:

  //medium.com is a protocol-relative URL, use explicit https:// instead


HTML-Proofer found 1 failure!


Error: Process completed with exit code 1.
```


## 기대 동작

build status Success
---

--- a/.github/workflows/pages-deploy.yml
+++ b/.github/workflows/pages-deploy.yml
@@ -75,6 +75,7 @@ jobs:
       url: ${{ steps.deployment.outputs.page_url }}
     runs-on: ubuntu-latest
     needs: build
+    if: github.ref == 'refs/heads/main'
     steps:
       - name: Deploy to GitHub Pages
         id: deployment

--- a/Gemfile
+++ b/Gemfile
@@ -14,5 +14,5 @@ end
 gem "wdm", "~> 0.2.0", :platforms => [:mingw, :x64_mingw, :mswin]
 
 group :jekyll_plugins do
-  gem "jekyll-linkpreview"
+  gem "jekyll-linkpreview", :git => "https://github.com/Rigu1/jekyll-linkpreview.git", :branch => "master"
 end

